docs/en/advanced_tutorials/config.md

```diff
@@ -1,3 +1,579 @@
+
+Here is an example of defining optimizers in a configuration file.
+
+`config_sgd.py`：
+
+```python
+optimizer = dict(type='SGD', lr=0.1, momentum=0.9, weight_decay=0.0001)
+```
+
+Suppose we have defined a registry `OPTIMIZERS`, which includes various optimizers, and then We can build the optimizer as below
+
+```python
+from mmengine import Config, optim
+from mmengine.registry import OPTIMIZERS
+
+import torch.nn as nn
+
+cfg = Config.fromfile('config_sgd.py')
+
+model = nn.Conv2d(1, 1, 1)
+cfg.optimizer.params = model.parameters()
+optimizer = OPTIMIZERS.build(cfg.optimizer)
+print(optimizer)
+```
+
+```
+SGD (
+Parameter Group 0
+    dampening: 0
+    foreach: None
+    lr: 0.1
+    maximize: False
+    momentum: 0.9
+    nesterov: False
+    weight_decay: 0.0001
+)
+```
+
+## Inheritance between configuration files
+
+Sometimes, the difference between two different configuration files is so small that only one field may be changed. However, we need to copy and paste everything, and only modify one line. It is not easy to locate the specific difference when we go back to this after a long time.
+
+In another case, multiple configuration files may have the same batch of fields, and we have to copy and paste them in different configuration files. It will also be hard to maintain these fields in a long time.
+
+### Overview of inheritance mechanism
+
+Here is an example to illustrate the inheritance mechanism.
+
+`optimizer_cfg.py`：
+
+```python
+optimizer = dict(type='SGD', lr=0.02, momentum=0.9, weight_decay=0.0001)
+```
+
+`resnet50.py`：
+
+```python
+_base_ = ['optimizer_cfg.py']
+model = dict(type='ResNet', depth=50)
+```
+
+Although we don't define `optimizer` in `resnet50.py`, since we wrote `_base_ = ['optimizer_cfg.py']`, it will inherit the fields defined in `optimizer_cfg.py`.
+
+```python
+cfg = Config.fromfile('resnet50.py')
+print(cfg.optimizer)
+```
+
+```
+{'type': 'SGD', 'lr': 0.02, 'momentum': 0.9, 'weight_decay': 0.0001}
+```
+
+`_base_` is a reserved field for the configuration file. It specifies the inherited(base) files for the current file. Inheriting multiple files will get all the fields at the same time, but it requires that there is no repeated fields defined in all base files.
+
+`runtime_cfg.py`：
+
+```python
+gpu_ids = [0, 1]
+```
+
+`resnet50_runtime.py`：
+
+```python
+_base_ = ['optimizer_cfg.py', 'runtime_cfg.py']
+model = dict(type='ResNet', depth=50)
+```
+
+At this case, reading the `resnet50_runtime.py` will give you 3 fields `model`, `optimizer`, and `gpu_ids`.
+
+```python
+cfg = Config.fromfile('resnet50_runtime.py')
+print(cfg.optimizer)
+```
+
+```
+{'type': 'SGD', 'lr': 0.02, 'momentum': 0.9, 'weight_decay': 0.0001}
+```
+
+In this way, we can disassemble the configuration file, define some general configuration files, and inherit them in the specific configuration file. This could avoid to define a lot duplicated contents in multiple configuration files.
```

Comment:
```suggestion
In this way, we can disassemble the configuration file, define some general configuration files, and inherit them in the specific configuration file. This could avoid defining a lot of duplicated contents in multiple configuration files.
```